drivers/clock_control/clock_control_nrf2_hfxo.c

```diff
@@ -48,6 +49,28 @@ static void hfxo_start_up_timer_handler(struct k_timer *timer)
 	}
 }
 
+static void request_hfxo(struct dev_data_hfxo *dev_data)
+{
+	unsigned int key;
+
+	key = irq_lock();
```

Comment:
since this function can be called by ZLI and by standard context don't you need a full lock here? For example function can be preempted during `request_count++` and counter will get compromised. Like we did in RTC timer: https://github.com/zephyrproject-rtos/zephyr/blob/3a86dfcdf68be62a922c49e21a5874c5d16ca2f2/drivers/timer/nrf_rtc_timer.c#L109

Reply:
the `irq_lock()` is defined as: https://github.com/zephyrproject-rtos/zephyr/blob/fcbf70f44d3c6d072288f8505bb1335faabdfb94/include/zephyr/irq.h#L252 that for arm cortex-m is implemented by https://github.com/zephyrproject-rtos/zephyr/blob/fcbf70f44d3c6d072288f8505bb1335faabdfb94/include/zephyr/arch/arm/asm_inline_gcc.h#L44-L73. so for CM33 is should disable all interrupts. I can change it anyway to similar behavior as in RTC.

Reply:
CM33 is actually ARMV& hence the `irq_lock()` and `irq_unlock()` doesn't provide enough locking level. Added similar API to one in RTC driver. In future I propose to add such API to be globally available.